jolt diff (do not merge)

ncitron opened the pull request.

ncitron committed:
add jolt toolchain

ncitron committed:
remove sys functions

ncitron committed:
fix allocator

ncitron committed:
add config

ncitron committed:
refactor allocator

ncitron committed:
remove commented out code

ncitron committed:
use jolt panic handler

ncitron committed:
add makefile

ncitron committed:
add archive make command

ncitron committed:
add build ci (#4)

* add ci

* run on pr

* specify branch

* use ref

* fix repo name

* use nightly toolchain

* use cargo native static

* fix ci build

* do a bad bad thing

* remove stage 2 ci check

* use ci branch

* add release

* disable gh action env

* fix

* fix archive

* fix ubuntu runner

* finalize

moodlezoup commented on .gitignore:
```diff
@@ -78,4 +76,5 @@ package.json
 tests/rustdoc-gui/src/**.lock
 
-# Before adding new lines, see the comment at the top.
+# Arhive
```
```suggestion
# Archive
```

moodlezoup commented on library/std/src/sys/pal/zkvm/mod.rs:
```diff
@@ -73,9 +72,11 @@ pub fn abort_internal() -> ! {
-        abi::sys_rand(buf.as_mut_ptr(), 4);
-    };
-    ((buf[0] as u64) << 32 + buf[1] as u64, (buf[2] as u64) << 32 + buf[3] as u64)
+    // let mut buf = [0u32; 4];
+    // unsafe {
+    //     abi::sys_rand(buf.as_mut_ptr(), 4);
+    // };
+    // (buf[0] as u64) << 32 + buf[1] as u64, (buf[2] as u64) << 32 + buf[3] as u64)
+
+    (0, 0)
```
Should we just
```suggestion
    unimplemented!("Unsupported");
```

ncitron replied:
No we don't want to do this as it will break hashmaps. The randomness if apparently used to to seed the hasher which prevents some weird worse-case performance of the hashmap implementation. Longer term we should see if we can feed in the fiat-shamir randomness here.

moodlezoup commented on library/std/src/lib.rs:
```diff
@@ -242,7 +242,7 @@
 #![allow(internal_features)]
 #![deny(rustc::existing_doc_keyword)]
-#![deny(fuzzy_provenance_casts)]
+// #![deny(fuzzy_provenance_casts)]
```
nit: 
```suggestion
```

moodlezoup commented on Makefile:
```diff
@@ -0,0 +1,13 @@
+build-toolchain:
+	GITHUB_ACTIONS=false CARGO_TARGET_RISCV32I_JOLT_ZKVM_ELF_RUSTFLAGS="-Cpasses=loweratomic" ./x build
+	GITHUB_ACTIONS=false CARGO_TARGET_RISCV32I_JOLT_ZKVM_ELF_RUSTFLAGS="-Cpasses=loweratomic" ./x build --stage 2
```
what's this `--stage 2` thing about?

ncitron replied:
Building the compiler happens in two stages. Tbh idk why but thats just how you build it.

ncitron committed:
use externally defined allocator

moodlezoup committed:
Fix build

moodlezoup committed:
God, are you there?

moodlezoup committed:
Fix checkout

moodlezoup committed:
Reduce space usage

moodlezoup committed:
feat: enable m extension (#5) (#8)

* feat: add m extension

* chore: rename target

Co-authored-by: Noah Citron <noah@jeff.org>

moodlezoup committed:
typo

WillPapper committed:
chore: Update toolchain to nightly-2024-09-30 (#10)

This matches the toolchain used by Jolt in this PR: https://github.com/a16z/jolt/pull/510

The compiler is updated to align with features used by Binius (in particular the use of `iter::repeat_n`)

sagar-a16z committed:
Add stable buulds and aarch64-unknown-linux-gnu to release  (#16)

* add aarch64-unknown-linux-gnu triple to release

* produce stable binaries
